### ISSUE TITLE
Wrong file copy

### ISSUE THREAD

**Issue by EmanueleCoppola** [kotarella1110/cordova-plugin-webpack#25]
It looks that running the command ```cordova build android -- --webpackConfig ./../../webpack.config.js``` the files are being copied before webpack finishes the build.
Example:
- First time we run the command (www empty and android platform not present).
- We add the android platform.
- We run the command, www folder is filled with the right files but `platforms/android/app/src/main/assets/www` empty.
- We rerun the command, `platforms/android/app/src/main/assets/www` is filled with the actual files, and files in www got replaced by newer ones (this will loop forever, everytime you re-run that command so you will never get the latest file into your application).

Is it because the prepare is beign executed before webpack finishes?
Or maybe, is it because I have some webpack plugins that are executed after the webpack build?
I tried to debug, but since the source code is "compiled" it's pretty hard to understand what promises are doing.

**Comment by EmanueleCoppola** [comment]
Running the same command as before and modifying the code like this: 

https://github.com/kotarella1110/cordova-plugin-webpack/blob/33be58f25fdb4174b2d98323f9cc5e24d98990ff/src/webpackCompile.ts#L21

```js
customWebpackConfig = webpackHelpers.webpackConfig(ctx.opts.projectRoot, argv.webpackConfig || argv.w);
compiler = webpack_1.default(customWebpackConfig);
console.log('START WEBPACK BUILD');
compiler.run(function (err, stats) {
    console.log('FINISH WEBPACK BUILD');
    if (err) {
        throw err;
    }
    console.log(stats.toString({
        chunks: false,
        colors: true,
    }));
});
```

this happens (in console output):
- START WEBPACK BUILD
- BUILD SUCCESSFUL
- FINISH WEBPACK BUILD

So it builds the application before the webpack task finishes.

**Comment by EmanueleCoppola** [comment]
I actually found out what the problem is.
There's a transpilation error in the plugin that comes from NPM.
I transpiled and installed it in my machine and it's working now, the JS code is totally different from the one downloaded from NPM.

**Comment by vitto32** [comment]
I can confirm the issue. 
The compiled `module.exports` on npm doesn't return a `Promise` but a sort of polyfill.
Cordova doesn't wait for it and starts the build in parallel with webpack `compiler.run`.

In HMR this is not noticeable (the compiled scripts are injected when webpack ends), but running a production build leads to unpredictable results.

**Comment by kotarella1110** [comment]
I recently noticed this issue.
In versions prior to v0.4.6, just as @EmanueleCoppola you said, the prepare is beign executed before webpack finishes.
But this issue has been fixed since [v0.4.7](https://github.com/kotarella1110/cordova-plugin-webpack/releases/tag/v0.4.7).
Rewriting the promise callback func from the callback func solved this problem.
Related commit: https://github.com/kotarella1110/cordova-plugin-webpack/commit/fd493bff575ee55741e75533bd8775496be26ec6
